src/server/plugins/plugin_collection.js

```diff
@@ -27,21 +39,26 @@ module.exports = class Plugins extends Collection {
     // clear the byIdCache
     this[byIdCache] = null;
 
-    for (let product of output) {
-
-      if (product instanceof api.Plugin) {
-        let plugin = product;
-        this.add(plugin);
-
-        let enabled = await plugin.readConfig();
-        if (!enabled) this.delete(plugin);
-        continue;
+    for (let plugin of output) {
+      if (!plugin instanceof api.Plugin) {
+        throw new TypeError('unexpected plugin export ' + inspect(plugin));
       }
 
-      throw new TypeError('unexpected plugin export ' + inspect(product));
+      await this.add(plugin);
+      if (!plugin.enabled) this.delete(plugin);
```

Comment:
It's really unclear why these two lines are necessary, I think it would be clearer if we added and removed the configuration from the config schema right here.